### ISSUE TITLE
Cannot Run Wordpress Flex App Engine Locally -- "No project ID was provided"

### ISSUE THREAD

**Issue by remy-actual** [GoogleCloudPlatform/php-docs-samples#337]
I've deployed Wordpress following the [Hosting WordPress in the App Engine Flexible Environment](https://cloud.google.com/php/tutorials/wordpress-app-engine-flexible) walk-through.  

The published site loads at https://competswim.appspot.com/ and I am able to login to the backend. 

Local connection to the CloudSQL instance via `cloud_sql_proxy` is good. 

However, running the local `wp server` command produces the following error on any browser request to http://localhost:8080:

```zsh
$ vendor/bin/wp server --path=wordpress
PHP 7.1.3 Development Server started at Thu Mar 30 09:46:28 2017
Listening on http://localhost:8080
Document root is /Users/remy/pro/src/google-cloud-platform/competswim/stg-competitiveswimmer-com/wordpress
Press Ctrl-C to quit.
# After making a http request to localhost:
PHP Fatal error:  Uncaught Google\Cloud\Core\Exception\GoogleException: No project ID was provided, and we were unable to detect a default project ID. in ~/stg-competitiveswimmer-com/vendor/google/cloud/src/Core/ClientTrait.php:170
Stack trace:
#0 ~/stg-competitiveswimmer-com/vendor/google/cloud/src/Core/ClientTrait.php(82): Google\Cloud\Storage\StorageClient->detectProjectId(Array)
#1 ~/stg-competitiveswimmer-com/vendor/google/cloud/src/Storage/StorageClient.php(84): Google\Cloud\Storage\StorageClient->configureAuthentication(Array)
#2 ~/stg-competitiveswimmer-com/wordpress/wp-config.php(24): Google\Cloud\Storage\StorageClient->__construct()
#3 ~/stg-competitiveswimmer-com/wordpress/wp-load.php(37): require_once('~/...')
#4 ~ in ~/stg-competitiveswimmer-com/vendor/google/cloud/src/Core/ClientTrait.php on line 170
^C
>>> elapsed time 18s
```

I generated the site with:  
`php wordpress-helper.php setup -n -d ./stg-competitiveswimmer-com --db_instance=CLOUDSQL_NAME --db_name=MYSQL_DB --db_user=MYSQL_USER -p PROJECT_ID --db_password='PASSKEY'`  

gcloud SDK is authenticated and my project is set as the default: 

```zsh
 ❯ gcloud config get-value project
Your active configuration is: [default]

competswim
```

How do I define the project ID in the local app?

**Comment by tmatsuo** [comment]
@remy-actual 

Thanks for the report. I think this is because the Cloud Storage Client library can not detect the Project ID. The current easiest workaround is to
- Create a service account and download the json key file
- Set an environment variable `GOOGLE_APPLICATION_CREDENTIALS`, pointing to that file

**Comment by remy-actual** [comment]
Thank you @tmatsuo, this resolved the issue.

**Comment by dovy** [comment]
Same issue, but this isn't working. Are specific permissions required for that account?

**Comment by tmatsuo** [comment]
@dovy 

I think "storage writer" role is required. Also you may need to add the service account to the bucket's ACL.

Or if you are lazy, you can give project editor's role.

**Comment by remy-actual** [comment]
@dovy Does `gcloud config list` show 
```shell
[core]
project = YOUR_PROJECT
``` 
?

Also you can specify the location of the local key file in `wp-config.php` with

```php
if (!$onGae) {
  putenv('GOOGLE_APPLICATION_CREDENTIALS=/Users/remy/project-sql-proxy.key.json');
}
```

**Comment by justinhartman** [comment]
@remy-actual I have the same issue as @dovy. I can confirm `gcloud config list` returns the correct project. I have a service account configured and have added the `putenv`, as you suggested, in my `wp-config.php` file to no avail.

**Comment by dovy** [comment]
I had to hardcore the proper path because pydatalab seems to use a different account file.

**Comment by jgacuca567** [comment]
@justinhartman When you deploy what, error do you get, please post a screenshot from your command line.

**Comment by dovy** [comment]
The issue is I have 2 accounts signed in on my computer. For whatever reason pydatalab takes only one of them, and doesn't expire the account. Pydatalab is only using the API.

I fear I've resolved the issue, so I can't see it. Basically it said I wasn't "authorized" because it wasn't picking up my current gcloud account, but the cached one.

**Comment by jgacuca567** [comment]
@dovy after running `gcloud init`, why don't you run one first and see which one is authorized.

**Comment by mikeheme** [comment]
I just ran into this issue and did manage to fix it with the advice in this thread but it took some trial and error. I followed the same google tutorial https://cloud.google.com/php/tutorials/wordpress-app-engine-flexible and the instructions just say "create a new service account" but doesn't tell you what role or permissions to use, initially I just gave my service account permissions to read/write Google SQL, which is why when I was trying to run the server locally I was getting the same error, even after exporting the env var GOOGLE_APPLICATION_CREDENTIALS with the path to my json key file. I ended up creating a new service account for operational purposes with access to everything in the project, I pointed the env var to the new json file and it worked.
